hoot-core/src/main/cpp/hoot/core/io/OsmApiChangesetElement.cpp

```diff
@@ -122,9 +123,18 @@ QString XmlElement::toTagString(const QXmlStreamAttributes& attributes) const
   QString buffer;
   QTextStream ts(&buffer);
   ts.setCodec("UTF-8");
+  QString key(attributes.value("k").toString());
   QString value(attributes.value("v").toString());
+  //  Tag values of length 255 need to be truncated
+  QString newValue(ApiTagTruncateVisitor::truncateTag(key, value));
+
   //  Make sure to XML encode the value
-  ts << "\t\t\t<tag k=\"" << attributes.value("k").toString() << "\" v=\"" << escapeString(value) << "\"/>\n";
+  ts << "\t\t\t<tag k=\"" << attributes.value("k").toString() << "\" v=\"";
+  if (newValue != "")
+    ts << escapeString(newValue);
+  else
+    ts << escapeString(value);
+  ts << "\"/>\n";
```

Comment:
While writing out changesets to a string to push to the OSM API, long tags are truncated here using the actual tag truncation code from the visitor.  Since the changeset isn't held in an `OsmMap` object, the visitor won't work.  It has to be done tag-by-tag, string-by-string.